client/src/App.jsx

```diff
@@ -1,14 +1,22 @@
 import { CssBaseline, ThemeProvider } from '@mui/material';
 import React from 'react';
+import { QueryClient, QueryClientProvider } from 'react-query';
+import { ReactQueryDevtools } from 'react-query/devtools'
+
 import Router from './pages/routing';
 import { theme } from 'theme';
 
+const queryClient = new QueryClient();
+
 const App = () => {
   return (
-    <ThemeProvider theme={theme}>
-      <CssBaseline />
-      <Router />
-    </ThemeProvider>
+    <QueryClientProvider client={queryClient}>
+      <ThemeProvider theme={theme}>
+        <CssBaseline />
+        <Router />
+      </ThemeProvider>
+      <ReactQueryDevtools initialIsOpen={false} />
+    </QueryClientProvider>
```

Comment:
QueryClientProvider powinien wrappować tylko ten komponent który faktycznie będzie z niego korzystał. 

```suggestion
      <ThemeProvider theme={theme}>
        <CssBaseline />
           <QueryClientProvider client={queryClient}>
             <Router />
             <ReactQueryDevtools initialIsOpen={false} />
           </QueryClientProvider>
      </ThemeProvider>
```